### DISCUSSION TITLE
Handling unmatched routes

### DISCUSSION THREAD

**Discussion by bombillazo** [nandorojo/solito#283] [Ideas]
Im using expo-router, and with their `<Link />` component if the route is not matched, it sends the user to a default 404 screen. So far with Solito, links that dont match simply do not navigate anywhere and log an error. 

It would be great to have the option to also route users to a route if it is unmatched.

**Comment by nandorojo** [comment]
That’s interesting. It should be possible with `SolitoProvider`’s middleware, where you pass a custom `useLinkTo` provided by expo router.

**Reply by nandorojo** [reply]
(That isn’t documented but it’s fairly simple)

**Reply by bombillazo** [reply]
Found the context files, but it would be great to learn more about how Solito scaffolds the context and how to pass options to it.

**Reply by nandorojo** [reply]
Basically, solito has an internal useLinkTo hook. By default, it calls useLinkTo from react navigation. However, you can override this hook at the provider level with a middleware.
